src/rules.c

```diff
@@ -521,12 +522,11 @@ void SchemaPrefixes_Free(TrieMap *t) {
   TrieMap_Free(t, freePrefixNode);
 }
 
-void SchemaPrefixes_Add(const char *prefix, StrongRef ref) {
-  size_t nprefix = strlen(prefix);
-  void *p = TrieMap_Find(SchemaPrefixes_g, (char *)prefix, nprefix);
+void SchemaPrefixes_Add(const char *prefix, size_t len, StrongRef ref) {
+  void *p = TrieMap_Find(SchemaPrefixes_g, (char *)prefix, len);
```

Comment:
Why should `find` be modifying the prefix? If it accepted `char*`, I would say that this is not safe. But it has been updated to be const-correct. So don't cast away const.

Reply:
Actually, `find` gets a `const char *`. Not sure why we cast here. removed